knora/dsplib/utils/xml_upload.py

```diff
@@ -304,6 +305,13 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
               f"{err.message}")
         quit(0)
 
+    if sys.platform.startswith("darwin") or sys.platform.startswith("linux"):
+        save_location = f"{os.path.expanduser('~')}/.dsp-tools"
+    elif sys.platform.startswith("win"):
+        save_location = "."
+    else:
+        save_location = "."
```

Comment:
Pathlib has a home directory that works OS independently. And generally I would not use any `os.path` code anymore, but use Pathlib instead

Reply:
I agree

Reply:
okay, it's noted with a TODO comment. For the moment, reading through the docs of Pathlib would be too time consuming. I want to take the time to do the transition to Pathlib properly, in order to really understand (and use) its advantages. 

In the current case, Pathlib wouldn't change anything, because I use the home directory only on Macs. On Windows, it' an entirely different story to obtain an app data storage location from the OS. 

But thanks for gently pushing me into using good libraries, I appreciate this endeavour!